pkg/validation/components_test.go

```diff
@@ -538,8 +552,10 @@ func TestValidateComponents(t *testing.T) {
 						assert.Regexp(t, tt.wantErr[i], merr.Errors[i].Error(), "Error message should match")
 					}
 				}
-			} else {
+			} else if tt.wantErr == nil {
 				assert.Equal(t, nil, err, "Error should be nil")
+			} else if tt.wantErr != nil {
+				assert.NotEqual(t, nil, err, "Error should not be nil")
```

Comment:
For clarity, it might make sense to do these checks first -- as it stands it looks like we check a subcase of `tt.wantErr != nil` first. Something like 

```go
if tt.wantErr == nil {
	assert.NoError(t, err, "Error should be nil")
	return
} else {
	if !assert.Error(t, err, "Should return error") {
		return
	}
}

merr, ok := err.(*multierror.Error)
if !ok {
	t.Fatalf("Unexpected error: %w", err)
}
if assert.Equal(t, len(tt.wantErr), len(merr.Errors), "Error list length should match") {
	for i := 0; i < len(merr.Errors); i++ {
		assert.Regexp(t, tt.wantErr[i], merr.Errors[i].Error(), "Error message should match")
	}
}
```

Reply:
updated